src/ray/core_worker/lib/golang/go_worker.cpp

```diff
@@ -0,0 +1,328 @@
+  auto contained_object_refs =
+      ray::core::CoreWorkerProcess::GetCoreWorker().GetObjectRefs(contained_object_ids);
+  std::shared_ptr<ray::Buffer> data = DataBufferToRayBuffer(in->data);
+  std::shared_ptr<ray::Buffer> meta = DataBufferToRayBuffer(in->meta);
+  return std::make_shared<ray::RayObject>(data, meta, contained_object_refs);
+}
+
+RAY_EXPORT int go_worker_SubmitActorTask(void *actor_id, char *method_name,
+                                         DataValue **input_values, int num_input_value,
+                                         int num_returns, void **object_ids) {
+  auto actor_id_obj = ByteArrayToId<ray::ActorID>((char *)actor_id);
+  std::vector<std::string> function_descriptor_list = {method_name};
+  ray::FunctionDescriptor function_descriptor =
+      ray::FunctionDescriptorBuilder::FromVector(ray::rpc::GOLANG,
+                                                 function_descriptor_list);
+  ray::core::RayFunction ray_function =
+      ray::core::RayFunction(ray::rpc::GOLANG, function_descriptor);
+  std::string name = "";
+  std::unordered_map<std::string, double> resources;
+  ray::core::TaskOptions task_options{name, num_returns, resources};
+  std::vector<std::unique_ptr<ray::TaskArg>> args;
+
+  for (size_t i = 0; i < num_input_value; i++) {
+    //    auto java_id = env->GetObjectField(arg, java_function_arg_id);
+    //    if (java_id) {
+    //      auto java_id_bytes = static_cast<jbyteArray>(
+    //          env->CallObjectMethod(java_id, java_base_id_get_bytes));
+    //      RAY_CHECK_JAVA_EXCEPTION(env);
+    //      auto id = JavaByteArrayToId<ObjectID>(env, java_id_bytes);
+    //      auto java_owner_address =
+    //          env->GetObjectField(arg, java_function_arg_owner_address);
+    //      RAY_CHECK(java_owner_address);
+    //      auto owner_address = JavaProtobufObjectToNativeProtobufObject<rpc::Address>(
+    //          env, java_owner_address);
+    //      return std::unique_ptr<TaskArg>(new TaskArgByReference(id, owner_address));
+    //    }
+    //    auto java_value =
+    //        static_cast<jbyteArray>(env->GetObjectField(arg, java_function_arg_value));
+    //    RAY_CHECK(java_value) << "Both id and value of FunctionArg are null.";
+    auto value = DataValueToRayObject(static_cast<DataValue *>(input_values[i]));
+    args.push_back(std::unique_ptr<ray::TaskArg>(new ray::TaskArgByValue(value)));
+  }
+  std::vector<ObjectID> obj_ids;
+  ray::core::CoreWorkerProcess::GetCoreWorker().SubmitActorTask(
+      actor_id_obj, ray_function, args, task_options, &obj_ids);
+
+  int object_id_size = ObjectID::Size();
+  for (size_t i = 0; i < obj_ids.size(); i++) {
+    void *result = malloc(object_id_size);
+    memcpy(result, (char *)obj_ids[i].Data(), object_id_size);
+    RAY_LOG(DEBUG) << "return object id:" << result;
+    object_ids[i] = result;
+  }
+  return 0;
+}
+
+DataBuffer *RayBufferToDataBuffer(const std::shared_ptr<ray::Buffer> buffer) {
```

Comment:
How do you ensure the lifetime of the buffer? Who owns the respective buffers and is responsible for destructing it?